src/main/java/games/strategy/engine/lobby/server/login/LobbyLoginValidator.java

```diff
@@ -155,10 +184,32 @@ private static String getBanDurationBreakdown(final Timestamp stamp) {
     return new BadWordController().list();
   }
 
-  private static String validatePassword(final Map<String, String> propertiesReadFromClient, final String clientName) {
+  private static String validatePassword(final Map<String, String> propertiesSentToClient,
+      final Map<String, String> propertiesReadFromClient, final String clientName) {
+    final String errorMessage = "Incorrect username or password";
     final UserDao userDao = new DbUserController();
+    final HashedPassword hashedPassword = userDao.getPassword(clientName);
+    if (hashedPassword == null) {
+      return errorMessage;
+    }
+    final String base64 = propertiesReadFromClient.get(ENCRYPTED_PASSWORD_KEY);
+    if (base64 != null) {
+      return decryptPassword(base64, propertiesSentToClient.get(RSA_PUBLIC_KEY), pass -> {
```

Comment:
Do I understand right?
- password is bcrypt encoded
- next we do sha512 which we decrypt on the server, presumably to mask the bcrypt password while transmitting.